DXMainClient/DXGUI/Multiplayer/GameLobby/CnCNetGameLobby.cs

```diff
@@ -1738,6 +1760,77 @@ private void HandleMapSharingBlockedMessage(string sender)
                 "they will be unable to participate in this match.".L10N("UI:Main:PlayerMissingMaDisabledSharing"), sender));
         }
 
+        /// <summary>
+        /// Download a map from CNCNet using a map hash ID.
+        ///
+        /// Users and testers can get map hash IDs from this URL template:
+        ///
+        /// - http://mapdb.cncnet.org/search.php?game=GAME_ID&search=MAP_NAME_SEARCH_STRING
+        ///
+        /// </summary>
+        /// <param name="parameters">
+        /// This is a string beginning with the sha1 hash map ID, and (optionally) the name to use as a local filename for the map file.
+        /// Every character after the first space will be treated as part of the map name.
+        ///
+        /// "?" characters are removed from the sha1 due to weird copy and paste behavior from the map search endpoint.
+        /// </param>
+        private void DownloadMapByIdCommand(string parameters)
+        {
+            string sha1;
+            string mapName;
+            string message;
+
+            // Check if the parameter's contain spaces.
+            // The presence of spaces indicates a user-specified map name.
+            parameters = parameters.Trim();
+            int firstSpaceIndex = parameters.IndexOf(' ');
+
+            if (firstSpaceIndex == -1)
+            {
+                // The user did not supply a map name.
+                sha1 = parameters;
+                mapName = "user_chat_command_download";
+            }
+            else
+            {
+                // User supplied a map name.
+                sha1 = parameters.Substring(0, firstSpaceIndex);
+                mapName = parameters.Substring(firstSpaceIndex + 1);
+                mapName = mapName.Trim();
+            }
+
+            // Remove erroneous "?" when someone double clicks and copies a map ID from the cncnet search endpoint.
+            // There is some weird whitespace that gets copied to chat as a "?" at the end. It's hard to spot, so just hold
+            // the user's hand.
+            sha1 = sha1.Replace("?", "");
+
+            GameModeMap loadedMap = GameModeMaps.Find(gmm => gmm.Map.SHA1 == sha1);
+            if (loadedMap != null)
+            {
+                message = String.Format(
+                    "The map for the ID \"{0}\" is already loaded from \"{1}.map\", delete the exiting file first before trying to download it again.",
```

Comment:
And here.

Reply:
updated all strings that were relevant to use L10N